pkg/util/azureclient/retry.go

```diff
@@ -88,7 +91,11 @@ func (rs *retrySender) Do(req *http.Request) (resp *http.Response, err error) {
 	}
 }
 
-func isRetryableError(clientName string, err error) bool {
+func isRetryableError(clientName, method string, err error) bool {
+	if method == "GET" && errors.IsMatchingSyscallError(err, syscall.ECONNRESET) {
```

Comment:
http.MethodGet preferred fwiw, but will lgtm anyway

Reply:
@asalkeld I'm not convinced this would catch #1593 - that looks like the NetError is wrapped by a autorest.DetailedError.  I suppose you can update IsMatchingSyscallError to unwrap?

Reply:
ok, i'll add the unwrapping to IsMatchingSyscallerrror so it can be reused.